### ISSUE TITLE
EmptyArrayAccess False Positive When Using Constants

### ISSUE THREAD

**Issue by mr-feek** [vimeo/psalm#3132]
### Description
👋 I've run into an issue regarding the usage of constants within class inheritance that seems to depend on the order of the variables declared.

## Failing
```php
<?php
abstract class A {
    public const KEYS = [];
    public const VALUES = [];
}

class B extends A {
    public const VALUES = ['there' => self::KEYS['hi']];
    public const KEYS = ['hi' => CONSTANTS::THERE];
}

class CONSTANTS {
    public const THERE = 'there';
}
```
If we remove the usage of `Constants::THERE` and instead hardcode `'there'`, the test case passes. Also, if we move the declaration of `B::VALUES` below `B::KEYS`, it passes also.

### Reproduce
https://psalm.dev/r/3163a9db2d

**Comment by muglug** [comment]
This is fixed in the latest release!

**Comment by mr-feek** [comment]
Hey @muglug ,

I can still reproduce the issue on psalm.dev on the latest commit in `master`  `d470903`

https://psalm.dev/r/3163a9db2d

> Psalm output (using commit d470903): 
> ERROR: EmptyArrayAccess - 8:39 - Cannot access value on empty array variable B::KEYS

**Comment by mr-feek** [comment]
Hey @muglug , I'm sure you get swamped with issues but it would be great to reopen this as it still seems to be failing on psalm.dev. 

Thanks!!

**Comment by mr-feek** [comment]
you the man! Thanks!
